designs/2022-community-eslint-org/README.md

```diff
@@ -0,0 +1,263 @@
+
+- It didn't receive an update in the last 12(?) months  
+  I think we can make an exception for packages that did get a PR merged which
+  was created by a member of the new org's "core" team (mostly for updating
+  compatibility with newer versions of ESLint).
+
+  We can also make an exception for packages that meet the other criteria, but
+  where the owner want to transfer ownership to the new org.  
+  I can for instance imagine that
+  [@not-an-aardvark](https://github.com/not-an-aardvark) would want to transfer
+  ownership of
+  [`eslint-plugin-eslint-plugin`](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin)
+  to the new org as well, as it's a recommended plugin for linting custom
+  plugins.  
+  <https://eslint.org/docs/developer-guide/working-with-plugins#linting>  
+  I can also imagine that people like [@gajus](https://github.com/gajus) at some
+  point wanting to transfer ownership of
+  [`eslint-plugin-flowtype`](https://github.com/gajus/eslint-plugin-flowtype) or
+  [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc) if they
+  would feel it would be better maintained by the community instead of putting
+  all workload on their shoulders.
+
+If a package is meeting these criteria, but we can't get into contact with the
+owner of the repo to transfer ownership over to the new org, we could
+temporarily fork the repo & publish the package as
+`@eslint-community-fork/<package name>` or so.
+
+Maybe [@G-Rath](https://github.com/G-Rath) &
+[@SimenB](https://github.com/SimenB) could give some insights on the acceptance
+criteria for `@jest-community` & we can learn from their findings as well.
+
+### What permissions should maintainers have?
+
+I think it would be best to have some sort of a "core" team (of at least 3(?)
+people) that's making the decisions in terms of acceptance to the repo, which
+would all be admin of the org.
+
+We can make a team (with sub-teams if we want) for each package, where we give
+admin rights on the repo for that specific package.
+
+The core team would help maintain all repos, whereas people in specific teams
+would only help maintain these repositories.
+
+**Bonus:** I think having our own place in the Discord server where we can
+discuss some things would be a good idea as well.  
+This way the "core" team has an open place to discuss some common things like
+extracting common functionality into a separate package, minimum Node & ESLint
+target, how to handle disputes between maintainers, ...  
+Having a channel for each separate package would maybe be a good idea as well,
+so the community has a change to ask questions to the maintainers as well.
+
+### What should be the role of the ESLint team?
+
+I think most important role would be to facilitate at least the contact with
+`@mysticatea`, so we can transfer his packages over to the `@eslint-community`
+org & gain `npm` publish rights for the packages.
```

Comment:
I mean the git tree of [weiran-zsd/eslint-plugin-node](https://github.com/weiran-zsd/eslint-plugin-node), not the npm package

Reply:
I would prefer to transfer the original repo to the new org and create new PRs to have all fixes from the fork upstream as well.

But that's only possible if we can get in touch with @mysticatea & transfer over the repo.

Reply:
Right.

In my priorities my number one is to "fix" https://github.com/weiran-zsd/eslint-plugin-node/issues/8 The merge with upstream is of less importance to me.

Hence this highlights that we/me/some of us are possibly conflating two projects here:

1. A place to collectively maintain forks like https://github.com/weiran-zsd/eslint-plugin-node
2. A neutral place to move unmaintained modules to

Right here and now we don't know when 2 can happen for anything but `eslint-plugin-promise`, we can only speculate, whereas 1. we can move ahead with right away. Moving ahead with 1. is outside the scope of this RFC though, but it's a useful distinction to keep in mind.

That would also mean that the `@eslint-community` very rarely would publish modules under the `@eslint-community` as it would never fork projects.

Reply:
Yes, I’m saying the ideal should be the original repo, and the fork’s repo should end up being deleted after its changes are pushed to the original.

Reply:
I agree with you - I’m saying that the ideal is to obviate the fork. Only if the original is truly impossible to recover should the existing fork be treated as canonical.

Reply:
I think we should question the value in moving rather than forking.

Projects like Babel and Fastify have both moved to prefixes for their non-forked modules. In a way it would make sense for an `@eslint-community` project to follow that and publish all modules under a prefix, eg: `@eslint-community/eslint-plugin-node`

If one anyway will do such a rename, then the only upsides of moving rather than forking is:

1. The ability to oneself deprecate the old name on npm rather than the original maintainer potentially doing so
2. An automated redirect from the old GitHub repo to the new (one which could be substituted somewhat by a link in the readme of that project instead)

While the downside with aiming for a move is:

1. Reliance on outside parties
2. Unclear when it will happen or if it at all will happen

I can see the elegance in moving rather than forking, but I think elegance carries little value here. (I'm talking about an actual fork rather than a GitHub fork)